app/server/lib/OIDCConfig.ts

```diff
@@ -155,14 +175,22 @@ export class OIDCConfig {
     return codeVerifier;
   }
 
-  private _makeUserProfileFromUserInfo(userInfo: UserinfoResponse) {
-      const email = userInfo.email;
-      const fname = userInfo.given_name ?? '';
-      const lname = userInfo.family_name ?? '';
-      return {
-        email,
-        name: `${fname} ${lname}`.trim(),
-      };
+  private _makeUserProfileFromUserInfo(userInfo: UserinfoResponse): Partial<UserProfile> {
+    return {
+      email: userInfo[ this._emailPropertyKey! ] as string,
+      name: this._extractName(userInfo)
+
+    };
+  }
+
+  private _extractName(userInfo: UserinfoResponse): string|undefined {
+    if (this._namePropertyKey) {
+      return userInfo[ this._namePropertyKey ] as string|undefined;
```

Comment:
Same here: `String(userInfo[this._namePropertyKey])` would be safer (in case the actual value happens to be of some other type).

Reply:
I rather propose `userInfo[ this._namePropertyKey ]?.toString()` so if the value is undefined/null, it is correctly handled here:

https://github.com/gristlabs/grist-core/blob/37dc1f8029173fc8a79e5dde24a603caa7ae7aca/app/gen-server/lib/HomeDBManager.ts#L690-L696